9 required changes (#80)
CaptainSifff/paper_teaching-learning-RSE: paper_teaching-learning-RSE.md

Some exemplary skills for which courses are already held are version control tools like git, HPC topics like multithreading, MPI and GPU computations, FAIR principles.

## Required Next Steps
We have identified the RSE as an individual who brings a unique set of skills for
supporting and undertaking modern digital research into research groups and teams.
We have elaborated on the basic competencies that RSEs may have, and we have
provided a detailed picture of how these competencies manifest along different
dimensions.
We have discussed who can deliver the necessary skills and the approaches to
support skills development. Now, in the final step, we want to discuss how this
specialization can become a natural part of the landscape of professions and which

> steps we need to take to professionalize their education.

Reviewer: I'm not quite sure I understood what is meant here. (the "Now, in the final step...." sentence). I think maybe I just found the sentence a little confusing to read so maybe it could be edited - I'm happy to do that once I clarify that I understand what is meant 🙂 

Which specialization is referred to? I assume RSE generally, but I wasn't sure if this was referring to some specific specialization within RSE. By I presume "landscape of professions" is just careers in general, i.e. not within the research technology space? Also I assume "their education" refers to the education of RSEs?

Reviewer 2: maybe "Now, in this final section" instead because required steps and final step somehow clash for me.

And I also agree with Jeremy that "this specialization" is a bit too vague here. I'd also say we should instead just write RSE somehow. (even though it has been in every sentence before)

A first qualification phase, possibly through academic structures, giving newcomers
to the field highly sought after professional skills, either directly from high school,
or as a result of a career change. But since the type of work undertaken by RSEs is
highly dynamic and often relies on knowledge about latest technologies, things change
quickly and an ongoing learning phase is also necessary to remain up-to-date with the
current developments, tools and techniques. 
In this context, we put forward the idea of having institutionalized structures that
facilitate continuous learning. These structures may be integrated within an academic
environment but they may also exist fully or partially outside of the academic sphere,
also providing opportunities for education on digital skills into wider society or
industry (see [@HPCAkademie] for an existing, HPC focused instance of this).

### First Professional Qualification
#### A possible graduation path within the classical university structures
We have put forward the idea that familiarity with research is a prerequisite for an RSE in order to be able to work effectively in the research space and in collaboration with researchers.

> To generate the required number of skilled persons
> We propose to offer a classical Bachelors and Master curriculum at select sites.
> In this particular example, we consider a path into RSE via a traditional university route involving Bachelors and Masters degree studies that include an RSE element.

Reviewer 3: Some typos in my opinion.

```suggestion
To generate the required number of skilled persons we propose to offer a classical Bachelor's and Master's curriculum at selected sites.
In this particular example, we consider a path into RSE via a traditional university route involving Bachelor's and Master's degree studies that include a RSE element.
```

Author: Thanks!

> However, we recognise that there are other routes into an RSE career and these are increasing.

Reviewer 3: ```suggestion
However, we recognise that there are other routes into a RSE career and these are increasing.
```

For example, some RSEs come from an industry background, others may come through apprenticeship or similar programmes.
In both cases, gaining knowledge of the research lifecycle and understanding the ways that researchers work towards solutions
to research challenges is something that can be developed on-the-job alongside training opportunities and the chance to work directly with researchers.
This leads naturally to the question, whether it is possible to become an RSE without a home research domain.
With software being a core element of the research process in so many different domains,
it is not helpful for everybody working in RSE to have a background in computing or software engineering.
Indeed, we consider it much more useful if new graduates looking to work in the RSE space come from a wide range of different domains.
Expertise, beyond software development skills,

> in another research domain can be an important element of an RSE team being able to support RSE projects in that domain.

Reviewer 3: ```suggestion
in another research domain can be an important element of a RSE team being able to support RSE projects in that domain.
```

Author: used as an abbreviation, RSE is vocalized starting with a vowel. Hence the liaison rules.

> A \${Domain}-RSE in the STEM domains can start with a Bachelor degree in their domain that shares large portions

Reviewer 3: ```suggestion
A \${Domain}-RSE in the STEM domains can start with a Bachelor's degree in their domain that shares large portions
```

of the curriculum with their original domain. There will only be some courses that advance peoples knowledge about digital
tools and how to create tools from them.
The Master's years bring them forward to actually applying their knowledge. So during these years they
simultaneously work on deepening their knowledge about team skills, a topic they have been neglecting so far,
they work towards getting to the current research in their domain, and getting up to speed about the current collaborative development practices.
So we can assume that the lectures to that point contain a mixture of domain specific content and RSE specific content

> (A good starting point for an RSE in the natural sciences), then we come to the question of the topic of the masters thesis.

Reviewer 3: ```suggestion
(A good starting point for an RSE in the natural sciences), then we come to the question of the topic of the master's thesis.
```

Of course special care has to be taken to integrate them at some point in their studies into mature projects
to expose them to the eperience of senior RSEs and how they approach problems.
An example of this \${DOMAIN} - RSE can be found in Appendix: [Bioinformatics skills and certification].
Of course, the next question for their future is whether a master's degree enables them to really be effective parts of a research group.
While we accept this is something of a generalisation, we argue that this is most likely not the case since
undertaking a PhD provides a much more extensive set of research training and experience that can be vital for a researcher's future.
Research environments differ internationally but in many cases there are formal barriers in the research landscape that require a PhD (e.g. eligibility for funding).
Hence a PhD is required to actively participate in science and hence we argue the regular RSE
should do a PhD that on one end combines knowledge of a research domain with software engineering heavy task
such that both pillars are suitably covered and have been exposed to research, software development and team dynamics.

> Having gone through this, this opens to them a future career as a lead RSE in their domain.

Reviewer 3: ```suggestion
Having gone through this, opens to them a future career as a lead RSE in their domain.
```

Author: Resolved.

Another would be to lay more effective foundations for future RSE work at a much earlier stage in undergraduate/postgraduate studies.
We see scope for establishing dedicated RSE Master's programmes which specialise in developing RSE skills and practices.
Some universities already offer dedicated master's programs in some domains.
Examples would be Computational Sciences in Engineering (CSE) or Bioinformatics
(see Appendix: [Bioinformatics skills and certification]).
Where appropriate similar programs should also be established in other domains.

#### Alternative first qualification paths
For completeness we mention that vocational programs like the MATSE [@MATSE] can provide a more application oriented less research oriented profile for individuals.
Since a certain amount of overlap with sciences is expected, it is expected that programs can be switched with minimal friction.

### Extra-academic institutions for continuous learning
As elaborated in the instruction the dynamic working place of an RSE requires continuous learning.
Knowledge-based industries have long recognized the value of qualifying their employees and hence invest time and money into this.

> We propose institutionalized structures that offer graduates a venue for life-long learning, or to deepn their skills.

Reviewer 3: ```suggestion
We propose institutionalized structures that offer graduates a venue for life-long learning, or to deepen their skills.
```

Centralizing this notion in a few institutions has some major benefits for society and industry as well,
since these become central hubs for offering the skills with digital tools.
The HPC-Akademie [@HPCAkademie] is a good example in the HPC space that offers its services to graduated

> researchers that want to deepen their HPC skills for research or to practitioners tht require them for industrial applications.

Reviewer 3: ```suggestion
researchers that want to deepen their HPC skills for research or to practitioners that require them for industrial applications.
```

> specializing the centres to certain skillsets enables them to effectively reach out into the sciences with their brand.

Reviewer 3: ```suggestion
Specializing the centres to certain skill sets enables them to effectively reach out into the sciences with their brand.
```
